src/components/ActionButton/ActionButton.tsx

```diff
@@ -71,12 +71,16 @@ const ActionButton = ({
   const [showLoader, setShowLoader] = useState(false);
   const [showSuccess, setShowSuccess] = useState(false);
   const ref = useRef<HTMLButtonElement>(null);
+  const loadingTimeTracker = useRef<number>(0);
+  const startLoadTime = useRef<Date | undefined>(undefined);
 
   // Set up loader timer
   useEffect(() => {
     let loaderTimeout: number;
 
     if (loading) {
+      // Keep track of time when loading is occured.
+      startLoadTime.current = new Date();
```

Comment:
This will run twice, so not capture the exact event of loading start. Because after `loading` state changes to true, we hit this code branch. Then just below call `setShowLoader`, which might change the `showLoader` state, which then would re-trigger the use effect. Not a big issue though as the re-render should be fast.